modules/misc/gtk.nix

```diff
@@ -166,14 +166,30 @@ in {
     home.packages = optionalPackage cfg.font ++ optionalPackage cfg.theme
       ++ optionalPackage cfg.iconTheme;
 
-    home.file.${cfg2.configLocation}.text =
-      concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
-      + cfg2.extraConfig;
+    home.file.${cfg2.configLocation} = {
+      text = concatStringsSep "\n" (mapAttrsToList formatGtk2Option ini) + "\n"
+        + cfg2.extraConfig;
+      onChange = let xsettingsd = "${pkgs.xsettingsd}/bin/xsettingsd";
+      in ''
+        DISPLAY=":0" $DRY_RUN_CMD ${xsettingsd} -c \
+        <( echo "Net/ThemeName \"${cfg.theme.name}\"" ) & \
+        ${pkgs.procps}/bin/pkill -f ${xsettingsd} || true
```

Comment:
```suggestion
        kill $?
         || true
```

Reply:
Awesome

Reply:
It seems killing it this way prevents xsettingsd from actually making the theme switch

Reply:
It seems to be a race condition, sometimes it works and sometimes not. I think adding a `sleep 0.1` might fix it

Reply:
Okay, it's probably some sort of race condition. Pkill probably takes just enough time to find the process to allow xsettingsd to do its thing, while `kill $!` is much faster. A `sleep 0.2` seems to fix it (0.1 is too low). Would this be a bad practice, i wonder?